File: NetFramework/C1.Win.Ribbon/VB/WordPad/BackstageTabs/SaveAsTab.vb
Public Class SaveAsTab

    Public Sub New()
        Initilize()
    End Sub

    Protected Overrides Sub OnMenuItemClick(ByVal mi As ListItem)
        Owner.SaveDocument(System.IO.Path.Combine(mi.Item.SubText, mi.Item.Text))
        mi.Item.LDate = DateTime.Now
    End Sub

    Protected Overrides Sub OnMenuButtonClick()
        Owner.SaveDocumentAs()
    End Sub

    Private Sub SaveAsTab_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Button.ItemImage = My.Resources.Save
        Button.ItemText = My.Resources.Browse_txt
        Caption = GetElementText(My.Resources.SaveAsTab_txt)
    End Sub
End Class

File: NetFramework/Ribbon/CS/WordPad/BackstageTabs/SaveAsTab.cs
using WordPad.BackstageTabs.Items;
using System;
using System.Drawing;

namespace WordPad.BackstageTabs
{
    public partial class SaveAsTab : BackstageTab
    {
        public SaveAsTab()
        {
            Button.ItemImage = Properties.Resources.Save;
            Button.ItemText = Properties.Resources.Browse_txt;
            Caption = GetElementText(Properties.Resources.SaveAsTab_txt);
        }

        #region override

        protected override void OnMenuItemClick(ListItem mi)
        {
            Owner.SaveDocument(System.IO.Path.Combine(mi.Item.SubText, mi.Item.Text));
            mi.Item.Date = DateTime.Now;
        }

        protected override void OnMenuButtonClick()
        {
            Owner.SaveDocumentAs();
        }

        #endregion  
    }
}
